pyiceberg/io/pyarrow.py

```diff
@@ -1745,14 +1747,42 @@ def write_file(table: Table, tasks: Iterator[WriteTask]) -> Iterator[DataFile]:
         key_metadata=None,
     )
 
-    if len(collected_metrics) != 1:
-        # One file has been written
-        raise ValueError(f"Expected 1 entry, got: {collected_metrics}")
-
     fill_parquet_file_metadata(
         data_file=data_file,
-        parquet_metadata=collected_metrics[0],
+        parquet_metadata=writer.writer.metadata,
```

Comment:
I checked this through the debugger, and this looks good. Nice change @jonashaag 👍

Reply:
You can also tell from the PyArrow code that it's identical :)